third_party/ibis/ibis_mssql/compiler.py

```diff
@@ -499,6 +499,14 @@ class MSSQLExprTranslator(alch.AlchemyExprTranslator):
             dt.Double: mssql.REAL,
             dt.String: mssql.VARCHAR,
             dt.Int64: mssql.MONEY,
+            # Unicode character strings
+            dt.String: mssql.NCHAR,
```

Comment:
I don't think this is right. Because this is an `update()` only one of the `dt.String` key value pairs will persist, the others will be overwritten. Did Ibis do this too?

Reply:
Ahhh I see what you mean, in Ibis code the mapping is reversed SQL Server Type to Ibis Data type: https://github.com/ibis-project/ibis/blob/master/ibis/backends/mssql/datatypes.py#L72. I just adapted to our implementation.

Reply:
We still have two entries for `dt.String`:

Which one survives will be open to chance:
```
            dt.String: mssql.VARCHAR,    <<<<<
            dt.Int64: mssql.MONEY,
            # Unicode character strings
            dt.String: mssql.NCHAR,      <<<<<
```

The Ibis code you linked to above only defines a key-value pair for `dt.String` once:
```
_MSSQL_TYPE_MAP = {
    dt.Boolean: mssql.BIT,
    dt.Int8: mssql.TINYINT,
    dt.Int16: mssql.SMALLINT,
    dt.Int32: mssql.INTEGER,
    dt.Int64: mssql.BIGINT,
    dt.Float16: mssql.FLOAT,
    dt.Float32: mssql.FLOAT,
    dt.Float64: mssql.REAL,
    dt.String: mssql.NVARCHAR,
}
```

It looks to me like those are used for mapping from SQLAlchemy types back to SQL Server types. Are you sure you need top change this bit?

Reply:
Yeah, you probably don't need to change this as long as you registered the data types in third_party/ibis/ibis_mssql/client.py